scripts/user_retirement/utils/thirdparty_apis/amplitude_api.py

```diff
@@ -36,6 +36,26 @@ def __init__(self, amplitude_api_key, amplitude_secret_key):
         self.base_url = "https://amplitude.com/"
         self.delete_user_path = "api/2/deletions/users"
 
+    @staticmethod
+    def get_instance(config):
```

Comment:
Configuration for each plugin API gets pushed down to the class, which is really where it belongs. For simplicity and backwards compatibility I've left all of the configuration variables the same.

This does involve passing the entire config object with potentially many sets of highly permissioned credentials into each plugin. That allows for things like plugins being able to contact the LMS or ecommerce, but isn't a security best practice. A potential improvement would be to move each plugin's configuration to a separate section and only pass that in. I've opted for backwards compatibility over all other considerations in this first pass.